src/openvr/OpenVRInterface.h

```diff
@@ -246,30 +258,37 @@ class OpenVRInterface : public ARVRInterface {
 
 	void blit(int p_eye, RID *p_render_target, Rect2 *p_rect) {
 		// blits out our texture as is, handy for preview display of one of the eyes that is already rendered with lens distortion on an external HMD
-		ARVRInterface::Eyes eye = (ARVRInterface::Eyes)p_eye;
+		XRInterface::Eyes eye = (XRInterface::Eyes)p_eye;
 		RID *render_target = (RID *)p_render_target;
 		Rect2 screen_rect = *(Rect2 *)p_rect;
 
-		if (eye == ARVRInterface::EYE_LEFT) {
+		if (eye == XRInterface::EYE_LEFT) {
 			screen_rect.size.x /= 2.0;
-		} else if (p_eye == ARVRInterface::EYE_RIGHT) {
+		} else if (p_eye == XRInterface::EYE_RIGHT) {
 			screen_rect.size.x /= 2.0;
 			screen_rect.position.x += screen_rect.size.x;
 		}
 
-		VSG::rasterizer->set_current_render_target(RID());
-		VSG::rasterizer->blit_render_target_to_screen(*render_target, screen_rect, 0);
+		/* THIS BIT NEEDS REDONE
+		RSG::rasterizer->set_current_render_target(RID());
+		RSG::rasterizer->blit_render_target_to_screen(*render_target, screen_rect, 0);
+		*/
+
 	}
 
-	int get_texid(RID *p_render_target) {
+	uint32_t get_texid(RID *p_render_target) {
 		// In order to send off our textures to display on our hardware we need the opengl texture ID instead of the render target RID
 		// This is a handy function to expose that.
-		RID *render_target = (RID *)p_render_target;
 
-		RID eye_texture = VSG::storage->render_target_get_texture(*render_target);
-		uint32_t texid = VS::get_singleton()->texture_get_texid(eye_texture);
+		// need to redo this
+
+	/*	RID *render_target = (RID *)p_render_target;
```

Comment:
getting the vulkan texture IDs are outside of my experience. Guidance appreciated

Reply:
I'm getting close, this function, when RID is retrieved and passed to compositor; return an invalid texture format error:

`VRCompositorError_TextureUsesUnsupportedFormat`

Judging by quick google search; looks like some work will need to be done to convert to proper format?

https://www.google.com/search?rlz=1C1CHBF_enCA923CA923&sxsrf=ALeKk031-USFLZnGy2JTFuO-8y-PTyFiLg%3A1609903644854&ei=HC71X-PbM8uItQWTgb7YCA&q=vulkan+textures+submit+openvr&oq=vulkan+textures+submit+openvr

```
	RID *render_target = (RID *)p_render_target;

		RID eye_texture = RSG::storage->render_target_get_texture(*render_target);

		return eye_texture.get_id();
		```

Reply:
On the bright side, it's submitting to the compositor and returning a sane error.

Reply:
Changing the texture to ` vr::TextureType_Vulkan` causes `Submit()` to crash. But I wholeheartedly believe im on the right track. Does it need to be a shared texture?

Reply:
So the problem here is that this works COMPLETELY differently with Vulkan. You're actually passing a lot more info about the render context instead of just the resulting texture.

We can go quick and dirty and just expose that information but I doubt any such change will be acceptable to merge into the core. you can have a look at whats needed for OpenVR and Vulkan here:
https://github.com/ValveSoftware/openvr/blob/master/samples/hellovr_vulkan/hellovr_vulkan_main.cpp#L1736

But it's better to redesign it from the core as we're now 3 years down the line and the VR world has changed a lot. One of the reasons we're giving OpenXR priority over the other interfaces as it design lends itself much better to implement improvements here.